Read size exceeds maximum object size

abrt build currently fails on Rawhide due to the following error:

```
In file included from /usr/include/features.h:503,
                 from /usr/include/sys/types.h:25,
                 from /usr/include/regex.h:23,
                 from ./../include/libabrt.h:12,
                 from check_recent_crash_file.c:18:
In function 'read',
    inlined from 'check_recent_crash_file' at check_recent_crash_file.c:37:14:
/usr/include/bits/unistd.h:38:10: error: '*read' specified size 18446744073709551614 exceeds maximum object size 9223372036854775807 [-Werror=stringop-overflow=]
   38 |   return __glibc_fortify (read, __nbytes, sizeof (char),
      |          ^~~~~~~~~~~~~~~
/usr/include/bits/unistd.h: In function 'check_recent_crash_file':
/usr/include/bits/unistd.h:26:16: note: in a call to function '*read' declared with attribute 'access (write_only, 2, 3)'
   26 | extern ssize_t __REDIRECT (__read_alias, (int __fd, void *__buf,
      |                ^~~~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [Makefile:677: libabrt_la-check_recent_crash_file.lo] Error 1
```

--- a/src/lib/check_recent_crash_file.c
+++ b/src/lib/check_recent_crash_file.c
@@ -17,18 +17,23 @@
 */
 #include "libabrt.h"
 
+#define MAX_EXECUTABLE_LENGTH 2048
+
 int check_recent_crash_file(const char *filename, const char *executable)
 {
     int fd = open(filename, O_RDWR | O_CREAT, 0600);
     if (fd < 0)
         return 0;
 
-    int ex_len = strlen(executable);
+    size_t ex_len = strlen(executable);
     struct stat sb;
     ssize_t sz;
 
     if (-1 == fstat(fd, &sb))
         perror_msg_and_die("Could not get information about opened file");
+
+    if (ex_len > MAX_EXECUTABLE_LENGTH)
+        perror_msg_and_die("Executable name too long, %zd > %d", ex_len, MAX_EXECUTABLE_LENGTH);
 
     if (sb.st_size != 0 /* if it wasn't created by us just now... */
         && (unsigned)(time(NULL) - sb.st_mtime) < 20 /* and is relatively new [is 20 sec ok?] */

